### PR TITLE
Terraform: Storage Pool Resource

### PR DESCRIPTION
**Pull request by vihangarao1**
# Description

- It includes Modifying Storage Pool Name
- Modify Media Type
- Set Read Ram Cache
- Enable/Disable RF Cache
- Added Integrated test cases for the API

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Integration Test

ss-storagepool

### CODE REVIEW

**Review by JacobGros — COMMENTED** [review]
please update description of PR

**Review by JacobGros — APPROVED** [review]

File: inttests/storagepool_test.go [hunk 1]
```diff
@@ -223,3 +223,62 @@ func TestCreateDeleteStoragePool(t *testing.T) {
...
+// Modify TestDisableRFCache
+func TestDisableRFCache(t *testing.T) {
+	domain := getProtectionDomain(t)
+	assert.NotNil(t, domain)
+	_, err := domain.DisableRFCache("b9b0be6400000003")
+	assert.Nil(t, err)
+}
+
+// Set TestSetRmcache
+func TestSetRmcache(t *testing.T) {
```
Comment (VamsiSiddu-7) [inline inttests/storagepool_test.go:270]: If i understood correct this function tests the modifyRmcache fn right? Please Rename it to TestModifyRmCache

File: inttests/storagepool_test.go [hunk 2]
```diff
@@ -223,3 +223,62 @@ func TestCreateDeleteStoragePool(t *testing.T) {
...
+	pool, _ := pd.FindStoragePool("", name, "")
+
+	// create a StoragePool instance to return
+	domain := goscaleio.NewStoragePoolEx(C, pool)
+
+	// create a storagePool via NewStoragePool to test
+	tempPool := goscaleio.NewStoragePool(C)
+	tempPool.StoragePool = pool
+
+	_ = domain.ModifyRMCache("true")
```
Comment (VamsiSiddu-7) [inline inttests/storagepool_test.go:283]: you can assert the err as nil here

**Review by VamsiSiddu-7 — CHANGES_REQUESTED** [review]
couple of review comments. Please address

File: inttests/storagepool_test.go (hunk 1 above)
Comment (vihangarao1) [reply inttests/storagepool_test.go]: Done

File: inttests/storagepool_test.go (hunk 2 above)
Comment (vihangarao1) [reply inttests/storagepool_test.go]: Done

**Review by khareRajshree — APPROVED** [review]

**Review by VamsiSiddu-7 — APPROVED** [review]

**Pull request merged**